### PR TITLE
Add piggy 1.5

### PR DESCRIPTION
**Pull request by Antonia-Chalka**
- [x] Updated licence file
- [x] Updated Readme file 
- [x] Dockerfile builds successfully
```
docker build -t staphb/piggy:1.5 .

[+] Building 0.1s (8/8) FINISHED
=> [internal] load build definition from Dockerfile 0.0s 
=> => transferring dockerfile: 744B 0.0s 
=> [internal] load .dockerignore 0.0s 
=> => transferring context: 2B 0.0s 
=> [internal] load metadata for docker.io/staphb/roary:3.12.0 0.0s 
=> [1/4] FROM docker.io/staphb/roary:3.12.0 0.0s 
=> CACHED [2/4] RUN apt-get -y update && apt-get -y install git 0.0s 
=> CACHED [3/4] RUN git clone https://github.com/harry-thorpe/piggy.git 0.0s 
=> CACHED [4/4] WORKDIR /data 0.0s 
=> exporting to image 0.0s 
=> => exporting layers 0.0s 
=> => writing image sha256:d508a4b28159bb7c4919d843d9f80fae89ed29ab8536a10900e91be736defd4b 0.0s 

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
```
- [x] Tool Runs
```
root@265d97582308:/data# docker run -it -v ${PWD}:/piggydata staphb/piggy:1.5 bash
root@265d97582308:/data# piggy -version
1.5
root@265d97582308:/data# roary -a
Use of uninitialized value in require at /usr/lib/x86_64-linux-gnu/perl/5.22/Encode.pm line 59.

Please cite Roary if you use any of the results it produces:
Andrew J. Page, Carla A. Cummins, Martin Hunt, Vanessa K. Wong, Sandra Reuter, Matthew T. G. Holden, Maria Fookes, Daniel Falush, Jacqueline A. Keane, Julian Parkhill,
"Roary: Rapid large-scale prokaryote pan genome analysis", Bioinformatics, 2015 Nov 15;31(22):3691-3693
doi: http://doi.org/10.1093/bioinformatics/btv421
Pubmed: 26198102

2021/07/12 19:47:36 Optional tool 'Rscript' not found in your $PATH
2021/07/12 19:47:36 Looking for 'awk' - found /usr/bin/awk
2021/07/12 19:47:36 Looking for 'bedtools' - found /usr/bin/bedtools
2021/07/12 19:47:36 Determined bedtools version is 2.25
2021/07/12 19:47:36 Looking for 'blastp' - found /usr/bin/blastp
2021/07/12 19:47:36 Determined blastp version is 2.2.31
2021/07/12 19:47:36 Looking for 'grep' - found /bin/grep
2021/07/12 19:47:36 Optional tool 'kraken' not found in your $PATH
2021/07/12 19:47:36 Optional tool 'kraken-report' not found in your $PATH
2021/07/12 19:47:36 Looking for 'mafft' - found /usr/bin/mafft
2021/07/12 19:47:36 Determined mafft version is 7.271
2021/07/12 19:47:36 Looking for 'makeblastdb' - found /usr/bin/makeblastdb
2021/07/12 19:47:36 Determined makeblastdb version is 2.2.31
2021/07/12 19:47:36 Looking for 'mcl' - found /usr/bin/mcl
2021/07/12 19:47:36 Determined mcl version is 14-137
2021/07/12 19:47:36 Looking for 'parallel' - found /usr/bin/parallel
2021/07/12 19:47:36 Determined parallel version is 20161222
2021/07/12 19:47:36 Looking for 'prank' - found /usr/bin/prank
2021/07/12 19:47:36 Determined prank version is 150803
2021/07/12 19:47:36 Looking for 'sed' - found /bin/sed
2021/07/12 19:47:36 Looking for 'cdhit' - found /usr/bin/cdhit
2021/07/12 19:47:36 Determined cdhit version is 4.6
2021/07/12 19:47:36 Looking for 'fasttree' - found /usr/bin/fasttree
2021/07/12 19:47:36 Determined fasttree version is 2.1
2021/07/12 19:47:36 Roary version 3.12.0
root@265d97582308:/data#
```
- [x] Tool has correct output
I've run it with a few *.gff files I had lying around. Output looks correct on first glance (correct folder types/structure etc), but I have not performed a 1:1 comparison with a non-docker piggy run

Notes:

- As this image contains an installation of roary (and roary seems to have been built with a specific version of roary) I would recommend running roary first then piggy with the same container.
- When running piggy, GNU parallel will keep outputting citation reminders that may clog up log files of a piggy run, but are a good reminder for a proper citation of all tools used in an analysis.

### CODE REVIEW

**Comment by kapsakcj** [comment]
Thanks for the PR! Everything looks great, but I have a few suggestions to help improve this dockerfile/docker image.

- `WORKDIR /data` is inheirited from the base image `staphb/roary:3.12.0`, and so the Piggy code is downloaded into `/data`. Could you please download the piggy code to the root directory `/` instead of `/data`? We typically reserve `/data` for passing data in and out of the container.
  - You can set `WORKDIR /` in the dockerfile prior to downloading the piggy code, which brings me to my next point...
- Could you please download the `piggy` code from the GitHub release tarball, instead of cloning the repo from master branch? It looks like there have not been any commits since the 1.5 release was created by the author. This will help ensure the docker image stays static. you will also need to install wget via apt-get `apt-get install wget`
```Dockerfile
RUN wget https://github.com/harry-thorpe/piggy/archive/refs/tags/v1.5.tar.gz && \
 tar -xzf v1.5.tar.gz && \
 rm v1.5.tar.gz
``` 
- If you make the above changes, the PATH will need to be updated as well to reflect the new location of the piggy executable `ENV PATH="/piggy-1.5/bin/:${PATH}" \`
- Lastly, at the very end of the dockerfile, reset the workdir to /data `WORKDIR /data`

Apologies if this is unclear, let me know if you have any questions.

I am happy to push commits and make these changes myself, just let me know.

Thanks for the PR! This is the first I've heard of Piggy, but it looks like an interesting & useful tool.

**Comment by Antonia-Chalka** [comment]
Thank you for the suggestions, I've made the requested changes (plus checked that piggy gets called successfully). Let me know if anything else is needed. 

And agreed! Piggy is definitely interesting, especially if you're after regulatory non-coding regions that get overlooked in pangenomic approaches.

File: piggy/1.5/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM staphb/roary:3.12.0
+
+# metadata
+LABEL base.image="staphb/roary:3.12.0"
+LABEL version="1"
```
Comment (kapsakcj) [inline piggy/1.5/Dockerfile:5]: please change line 5 to `LABEL dockerfile.version="1"`

explanation here: https://github.com/StaPH-B/docker-builds/issues/118

**Comment by kapsakcj** [comment]
Thank you for making those changes, I appreciate it. This will help to keep things reproducible in the future.

I made one small comment on a LABEL and after you push that change, I will merge in to the master branch and build on dockerhub & quay.

**Comment by Antonia-Chalka** [comment]
All done  👍

**Comment by kapsakcj** [comment]
alrighty, the image is now available on dockerhub and quay. Enjoy and thanks again for contributing!

https://hub.docker.com/r/staphb/piggy

https://quay.io/repository/staphb/piggy?tab=tags

**Pull request merged**